src/main/java/com/philips/research/bombase/core/nuget/domain/NugetClient.java

```diff
@@ -0,0 +1,105 @@
+/*
+ * Copyright (c) 2020-2021, Koninklijke Philips N.V., https://www.philips.com
+ * SPDX-License-Identifier: MIT
+ */
+
+package com.philips.research.bombase.core.nuget.domain;
+
+import com.fasterxml.jackson.annotation.JsonAutoDetect;
+import com.fasterxml.jackson.annotation.PropertyAccessor;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.DeserializationFeature;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.PropertyNamingStrategies;
+import com.fasterxml.jackson.dataformat.xml.XmlMapper;
+import com.github.packageurl.PackageURL;
+import com.philips.research.bombase.core.meta.PackageMetadata;
+import com.philips.research.bombase.core.nuget.NugetException;
+import org.springframework.stereotype.Component;
+import retrofit2.Call;
+import retrofit2.Retrofit;
+import retrofit2.converter.jackson.JacksonConverterFactory;
+
+import java.io.IOException;
+import java.net.URI;
+import java.util.Locale;
+import java.util.Optional;
+
+@Component
+public class NugetClient {
+    // TODO: Can these mappers be cleaned up a bit more?
+    private static final ObjectMapper MAPPER = new ObjectMapper()
+            .configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, false)
+            .setVisibility(PropertyAccessor.FIELD, JsonAutoDetect.Visibility.NON_PRIVATE)
+            .setPropertyNamingStrategy(PropertyNamingStrategies.SNAKE_CASE);
+    private static final ObjectMapper XMLMAPPER = new XmlMapper()
+            .configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, false)
+            .setVisibility(PropertyAccessor.FIELD, JsonAutoDetect.Visibility.NON_PRIVATE);
+    private final NugetAPI restJson;
+    private final NugetAPI restXml;
+    private final URI baseURI;
+
+    NugetClient() {
+        this(URI.create("https://api.nuget.org/v3/"));
+    }
+
+    NugetClient(URI uri) {
+        final var retrofitJson = new Retrofit.Builder()
+                .baseUrl(uri.toASCIIString())
+                .addConverterFactory(JacksonConverterFactory.create(MAPPER))
+                .build();
+        final var retrofitXML = new Retrofit.Builder()
+                .baseUrl(uri.toASCIIString())
+                .addConverterFactory(JacksonConverterFactory.create(XMLMAPPER))
+                .build();
+        restJson = retrofitJson.create(NugetAPI.class);
+        restXml = retrofitXML.create(NugetAPI.class);
+        baseURI = uri;
+    }
+
+    Optional<PackageMetadata> getPackageMetadata(PackageURL purl) {
+        final var catalogResponse =
+                query(restJson.getCatalogEntry(purl.getName().toLowerCase(
+                        Locale.ROOT), purl.getVersion()));
+        if (catalogResponse.isEmpty() || catalogResponse.get().catalogEntry == null) {
+            return Optional.empty();
+        } else {
```

Comment:
This `else` statement does not do anything, so it only introduces visual clutter. (Remove it?)

The concept of "guard conditions" is to exit early for special cases, and otherwise continue with the normal flow of the function. Using `else` in these cases only compresses the remaining code due to the indent, which reduces readability.

Reply:
You're right. Not sure why I did not remove them earlier. I usually get warnings for leaving them...